### PR TITLE
Mimic desktop behaviour for FileSave/FileOpen dialogs: focus and submitting by keyboard

### PR DESCRIPTION
**Pull request by ivoras**
### Description:
Two commits, each introducing one change:

1. Submitting (clicking ENTER) while in the fileName Entry the SaveDialog submits the form (clicks the Save button)
2. When the Save / Open file dialog is shown, the fileName entry is focused immediately

This mimics the native behaviour on desktop environments (at least Windows & Linux, can't test on OSX).

### Checklist:

- [ ] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

File: dialog/file.go [hunk 1]
```diff
@@ -644,6 +648,13 @@ func showFile(file *FileDialog) *fileDialog {
 
 	d.setLocation(file.effectiveStartingDir())
 	d.win.Show()
+	if file.save {
+		// Focus on the filename Entry as soon as the dialog is shown
+		go func() {
+			time.Sleep(25 * time.Millisecond)
+			file.dialog.win.Canvas.Focus(file.dialog.fileName.(*widget.Entry))
+		}()
```
Comment (Jacalz) [inline dialog/file.go:656]: You can focus things before they are shown as long as the items are part of the window contents. Spinning up a goroutine and sleeping for a random amount of time seems like a hacky workaround.

**Review by Jacalz — CHANGES_REQUESTED** [review]
Thanks for these qol improvements. I left a comment inline about cleaning up the code a bit. Also, would you mind giving the PR a descriptive title?

File: dialog/file.go (hunk 1 above)
Comment (ivoras) [reply dialog/file.go]: Done. I see why.

Comment (Jacalz) [reply dialog/file.go]: Remove the comment as well. It no longer valid (technically wasn't 100% valid before either).

Comment (ivoras) [reply dialog/file.go]: Done

**Review by Jacalz — APPROVED** [review]
Thanks for the nice QOL improvements :+1:

**Pull request merged**